src/metax_api/models/catalog_record.py

```diff
@@ -372,7 +372,20 @@ def user_has_access(self, request):
             return True
 
         elif request.method in READ_METHODS:
-            return True
+            state = self.state
+            owner = self.metadata_provider_user
+            if request.user.username is None: # unauthenticated user
+                if state == 'published':
+                    return True
+                else:
+                    raise Http404
+            else: # enduser
+                if state == 'published':
+                    return True
+                elif state == 'draft' and owner == request.user.username:
+                    return True
+                else:
+                    raise Http404
```

Comment:
The above looks clear and simple :+1: 

I would only avoid using these "mid step" variables ``state`` and ``owner``, and instead use ``self.state`` and ``self.metadata_provider_user`` directly.

Additionally, instead of using the strings ``'published'`` and ``'draft'``, use variables ``self.STATE_PUBLISHED`` and ``self.STATE_DRAFT`` which were defined previously as accepted values for ``self.state``. This way, if those values ever change for whatever reason, this part of the code does not need to be changed.